src/components/core/pivotTable/AggregationsSubMenu.tsx

```diff
@@ -0,0 +1,112 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import { AFM, Execution } from '@gooddata/typings';
+import { Header, Item, ItemsWrapper } from '@gooddata/goodstrap/lib/List/MenuList';
+
+import SubMenu from '../../menu/SubMenu';
+import { IMenuAggregationClickConfig } from '../../../interfaces/PivotTable';
+import { IColumnTotal } from './AggregationsMenu';
+
+const MENU_HEADER_OFFSET = -36;
+
+export interface IAggregationsSubMenuProps {
+    intl: ReactIntl.InjectedIntl;
+    totalType: AFM.TotalType;
+    toggler: JSX.Element;
+    isMenuOpened?: boolean;
+    rowAttributeHeaders: Execution.IAttributeHeader[];
+    measureLocalIdentifiers: string[];
+    columnTotals: IColumnTotal[];
+    onAggregationSelect: (clickConfig: IMenuAggregationClickConfig) => void;
+}
+
+export default class AggregationsSubMenu extends React.Component<IAggregationsSubMenuProps> {
+    public static defaultProps: Partial<IAggregationsSubMenuProps> = {
+        isMenuOpened: false
+    };
+
+    public render() {
+        const { toggler, isMenuOpened, intl } = this.props;
+        const menuOpenedProp = isMenuOpened ? { opened: true } : {};
+
+        return (
+            <SubMenu
+                toggler={toggler}
+                offset={MENU_HEADER_OFFSET}
+                {...menuOpenedProp}
+            >
+                <ItemsWrapper>
+                    <div className="s-table-header-submenu-content">
+                        <Header>{intl.formatMessage({ id: 'visualizations.menu.aggregations.rows' })}</Header>
+                        {this.renderSubMenuItems()}
+                    </div>
+                </ItemsWrapper>
+            </SubMenu>
+        );
+    }
+
+    private getAttributeLocalIdentifier(
+        rowAttributeHeaders: Execution.IAttributeHeader[],
+        headerIndex: number
+    ): string {
+        return rowAttributeHeaders[headerIndex].attributeHeader.localIdentifier;
+    }
+
+    private getAttributeName(
+        rowAttributeHeaders: Execution.IAttributeHeader[],
+        headerIndex: number
+    ): string {
+        return headerIndex === 0
+            ? this.props.intl.formatMessage({ id: 'visualizations.menu.aggregations.all-rows' })
+            : rowAttributeHeaders[headerIndex - 1].attributeHeader.name;
```

Comment:
Maybe add some comment or extract this branch to a method to explain why we do this `index - 1` magic?